eth/catalyst/api_types.go

```diff
@@ -21,50 +21,82 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
-//go:generate go run github.com/fjl/gencodec -type assembleBlockParams -field-override assembleBlockParamsMarshaling -out gen_blockparams.go
+//go:generate go run github.com/fjl/gencodec -type AssembleBlockParams -field-override assembleBlockParamsMarshaling -out gen_blockparams.go
 
-// Structure described at https://hackmd.io/T9x2mMA4S7us8tJwEB3FDQ
-type assembleBlockParams struct {
-	ParentHash common.Hash `json:"parentHash"    gencodec:"required"`
-	Timestamp  uint64      `json:"timestamp"     gencodec:"required"`
+// Structure described at https://github.com/ethereum/execution-apis/pull/74
+type AssembleBlockParams struct {
+	ParentHash   common.Hash    `json:"parentHash"    gencodec:"required"`
+	Timestamp    uint64         `json:"timestamp"     gencodec:"required"`
+	Random       common.Hash    `json:"random"        gencodec:"required"`
+	FeeRecipient common.Address `json:"feeRecipient"  gencodec:"required"`
 }
 
 // JSON type overrides for assembleBlockParams.
 type assembleBlockParamsMarshaling struct {
 	Timestamp hexutil.Uint64
 }
 
-//go:generate go run github.com/fjl/gencodec -type executableData -field-override executableDataMarshaling -out gen_ed.go
+//go:generate go run github.com/fjl/gencodec -type ExecutableData -field-override executableDataMarshaling -out gen_ed.go
 
-// Structure described at https://notes.ethereum.org/@n0ble/rayonism-the-merge-spec#Parameters1
-type executableData struct {
-	BlockHash    common.Hash    `json:"blockHash"     gencodec:"required"`
-	ParentHash   common.Hash    `json:"parentHash"    gencodec:"required"`
-	Miner        common.Address `json:"miner"         gencodec:"required"`
-	StateRoot    common.Hash    `json:"stateRoot"     gencodec:"required"`
-	Number       uint64         `json:"number"        gencodec:"required"`
-	GasLimit     uint64         `json:"gasLimit"      gencodec:"required"`
-	GasUsed      uint64         `json:"gasUsed"       gencodec:"required"`
-	Timestamp    uint64         `json:"timestamp"     gencodec:"required"`
-	ReceiptRoot  common.Hash    `json:"receiptsRoot"  gencodec:"required"`
-	LogsBloom    []byte         `json:"logsBloom"     gencodec:"required"`
-	Transactions [][]byte       `json:"transactions"  gencodec:"required"`
+// Structure described at https://github.com/ethereum/execution-apis/pull/74/files
+type ExecutableData struct {
+	BlockHash     common.Hash    `json:"blockHash"     gencodec:"required"`
+	ParentHash    common.Hash    `json:"parentHash"    gencodec:"required"`
+	Coinbase      common.Address `json:"coinbase"      gencodec:"required"`
+	StateRoot     common.Hash    `json:"stateRoot"     gencodec:"required"`
+	ReceiptRoot   common.Hash    `json:"receiptsRoot"  gencodec:"required"`
```

Comment:
```suggestion
	ReceiptRoot   common.Hash    `json:"receiptRoot"  gencodec:"required"`
```
I believe this should be `receiptRoot`, not `receiptsRoot`.

Reply:
https://github.com/ethereum/execution-apis/blob/main/src/engine/interop/specification.md#executionpayload